deliveries/ink_contract_multi-sig-1.md

```diff
@@ -0,0 +1,39 @@
+# Milestone Delivery :mailbox:
+
+**The invoice has been send out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/smart-contract-bounty/Support-Docs/blob/master/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** [Ink Multisig](https://github.com/use-inkubator/Ecosystem-Grants/blob/master/applications/ink_contract_multi-sig.md)
+
+**Context**
+
+The provided smart contract is a Multi-Signature (MultiSig) contract implemented in ink! - a Rust-based eDSL for writing smart contracts for the Polkadot ecosystem. A MultiSig contract is a type of contract that requires approval from multiple accounts (owners) before a transaction is executed.
+
+Key details and deliverables of this smart contract are as follows:
+
+1) Multisig Contract: The contract enables a group of owners to propose, approve, or reject transactions. Transactions are only executed if an approval threshold is met.
+
+2) Threshold Management: The smart contract includes functionality for changing the approval threshold - the minimum number of approvals needed for a transaction to be executed.
+
+3) Owner Management: The contract also has functions for adding and removing owners, as well as checking if a specific account is an owner. The number of owners is capped by a predefined constant (MAX_OWNERS).
+
+4) Transaction Management: Each transaction is stored with a unique TxId and includes the recipient's address, a function selector, input data, transferred value, gas limit, and a boolean that controls whether reentry is allowed. Once a transaction is proposed, the owners can approve or reject it.
+
+The following documents delivered in the docs folder provide more context about our research and implementation:
+
+1) Overcoming Reentrancy and Circular Call Challenges in Polkadot Smart Contracts

```

Comment:
Same here, I don't understand who this was written for

Reply:
During the POC phase, we discovered an intriguing issue within Polkadot smart contracts. As a result, we created this document to explain the nature of this problem and presented a solution to ensure that the smart contracts operate as intended. This resource is not only valuable for our project but also has the potential to assist other developers facing similar challenges in their work.